packages/cozy-client/src/queries/dsl.js

```diff
@@ -337,6 +337,23 @@ class QueryDefinition {
  */
 export const Q = doctype => new QueryDefinition({ doctype })
 
+/**
+ * Check if the query is a getById() query
+ *
+ * @param {QueryDefinition} queryDefinition The query definition
+ *
+ * @returns {boolean}
+ */
+export const isAGetByIdQuery = queryDefinition => {
+  if (
+    queryDefinition &&
+    queryDefinition.id &&
+    !queryDefinition.ids &&
+    !queryDefinition.selector
+  )
```

Comment:
Nit: Wonder why we have `selector` and not other attributes here ? I think the code works but maybe it could be more generic with `Object.keys(queryDefinition).length === 1 && queryDefinition.id !== undefined` ?

Reply:
Ah nice way to detect that! I added selector since there is nothing that block us to do getById('1').where()... I just started by selector and ids and wanted to add other attributes but I forgot. 

Will give a shot to the direction you said!

Reply:
We set all the attributes to undefined even if we don't use them. So I filtered the undefined value and check if the length is 2. 👍